File: Cpp/Remove Duplicates from Sorted Array II.cpp
class Solution {
public:
    /**
     * @param A: a list of integers
     * @return : return an integer
     */
    int removeDuplicates(vector<int> &nums) {
        // write your code here
        if (nums.size() <= 2) return nums.size();
        int index = 0, res = 1;
        for (int i = 1; i < nums.size(); i ++) {
            if (nums[i] != nums[index]) {
                nums[++index] = nums[i];
                res = 1;
            } else {
                if (res < 2) {
                    res += 1;
                    nums[++index] = nums[i];
                }
            }
        }
        return index + 1;
    }
};

File: Java/Remove-Duplicates-from-Sorted-Array-II.java
public class Solution {
    /**
     * @param A: a array of integers
     * @return : return an integer
     */
    public int removeDuplicates(int[] nums) {
        // write your code here
        int cur = -1, time = 1;
        for (int i = 0; i < nums.length; i ++) {
        	if (cur == -1 || nums[i] != nums[cur]) {
        		time = 1;
        		nums[++cur] = nums[i];
        	} else if (nums[i] == nums[cur]) {
        		if (time++ == 1) {
        			nums[++cur] = nums[i];
        		}
        	}
        }
        return cur + 1;
    }
}
